napari/layers/shapes/shapes.py

```diff
@@ -297,6 +324,43 @@ class Shapes(Layer):
     # in the thumbnail
     _max_shapes_thumbnail = 100
 
+    _drag_modes = {
+        Mode.SELECT: select,
+        Mode.DIRECT: select,
+        Mode.VERTEX_INSERT: vertex_insert,
+        Mode.VERTEX_REMOVE: vertex_remove,
+        Mode.ADD_RECTANGLE: add_rectangle,
+        Mode.ADD_ELLIPSE: add_ellipse,
+        Mode.ADD_LINE: add_line,
+        Mode.ADD_PATH: add_path_polygon,
+        Mode.ADD_POLYGON: add_path_polygon,
+    }
+
+    _move_modes = {
+        Mode.SELECT: highlight,
+        Mode.DIRECT: highlight,
+        Mode.VERTEX_INSERT: highlight,
+        Mode.VERTEX_REMOVE: highlight,
+        Mode.ADD_PATH: add_path_polygon_creating,
+        Mode.ADD_POLYGON: add_path_polygon_creating,
+    }
+    _cursor_modes = {
+        Mode.PAN_ZOOM: 'standard',
+        Mode.SELECT: 'pointing',
+        Mode.DIRECT: 'pointing',
+        Mode.VERTEX_INSERT: 'cross',
+        Mode.VERTEX_REMOVE: 'cross',
+        Mode.ADD_RECTANGLE: 'cross',
+        Mode.ADD_ELLIPSE: 'cross',
+        Mode.ADD_LINE: 'cross',
+        Mode.ADD_PATH: 'cross',
+        Mode.ADD_POLYGON: 'cross',
+    }
```

Comment:
We could also add other checks, like that cursors are defined for all modes.

Reply:
I ended up doing this as some test were failing I introduced a no_op function and bind it when nothing should be done.

Reply:
sounds good